nemo/collections/tts/torch/tts_tokenizers.py

```diff
@@ -282,6 +282,9 @@ def __init__(
              Note that lower() function shouldn't applied here, because text can contains phonemes (it will be handled by g2p).
```

Comment:
Current implementation of `EnglishPhonemesTokenizer` with `phoneme_probability > 0` in G2P also mixes representation even in eval mode. We don't need it, right? Let's add flag to `EnglishPhonemesTokenizer.encode` like `eval_mode`. Then in `encode` method you need to replace this line `g2p_text = self.g2p(text)` with something like that `g2p_text = self.g2p(text, mix_representation=False if eval_mode else None)`. And then you need to add `mix_representation` in g2p and handle it. Also we need to update https://github.com/blisc/NeMo/blob/tts_finetuning_updates/nemo/collections/tts/models/fastpitch.py#L197 somehow and, please, also update Mixer-TTS function (should be the same changes).

Reply:
Do you want it to be run in eval()?
I can enable for train, and disable for infer.

Reply:
I want to have ability to turn off stochastic mixed representation if I need it. I can need it during evaluation_step(), also I can need it in `def self.parse()`…

Reply:
It is not good that our metrics in validation step will be depend on `phoneme_probability`, because they will be not representative for user (in inference user will use `parse` and it produces deterministic output). 

How to fix it? 

I see two variants:

1. In `__setup_dataloader_from_config` for `name="val"` we can do something like that:

 ```
 val_vocab = copy.deepcopy(self.vocab) # or any other alternative to copy.deepcopy
 val_vocab. phoneme_probability = 1.0
 val_vocab.g2p. phoneme_probability = 1.0 # need to check if g2p is
 dataset = instantiate(...text_tokenizer=val_vocab)
 ```
 
 We need to do the same for Mixer-TTS.
 
 2. We can add field `self.dataset_mode` to TTSDataset  (it can be "val_dataset" or "train_dataset"). In this case, we can add your context manager if `self.dataset_mode == val_dataset`

Reply:
Added a fix to fastpitch. Please update mixerTTS as you want